test/src/main/java/org/apache/accumulo/test/upgrade/GCUpgrade9to10TestIT.java

```diff
@@ -118,38 +113,70 @@ public void gcUpgradeNoDeletesIT() throws Exception {
+    int numberOfEntries = 100_000;
+    String longpathname = StringUtils.repeat("abcde", 100);
+    assertEquals(500, longpathname.length());
+
+    // dynamically get the batch size from the other class
+    Field CANDIDATE_BATCH_SIZE_field = Upgrader9to10.class.getDeclaredField("CANDIDATE_BATCH_SIZE");
+    CANDIDATE_BATCH_SIZE_field.setAccessible(true);
+    long CANDIDATE_BATCH_SIZE = CANDIDATE_BATCH_SIZE_field.getLong(null);
+    // sanity check to ensure that any batch size assumptions are still valid in this test
+    assertEquals(4_000_000, CANDIDATE_BATCH_SIZE);
+
+    // ensure test quality by making sure we have enough candidates to
+    // exceed the batch size at least ten times
+    long numBatches = numberOfEntries * longpathname.length() / CANDIDATE_BATCH_SIZE;
+    assertTrue("Expected numBatches between 10 and 15, but was " + numBatches,
+        numBatches > 10 && numBatches < 15);
+
     Ample.DataLevel level = Ample.DataLevel.USER;
 
     log.info("Filling metadata table with lots of bogus delete flags");
     try (AccumuloClient c = Accumulo.newClient().from(getClientProperties()).build()) {
-      addEntries(c, level.metaTable(), somebignumber, longpathname);
+      Map<String,String> expected = addEntries(c, level.metaTable(), numberOfEntries, longpathname);
+      assertEquals(numberOfEntries + numberOfEntries / 10, expected.size());
+
+      Range range = MetadataSchema.DeletesSection.getRange();
 
       sleepUninterruptibly(1, TimeUnit.SECONDS);
+      try (Scanner scanner = c.createScanner(level.metaTable(), Authorizations.EMPTY)) {
+        Map<String,String> actualOldStyle = new HashMap<>();
+        scanner.setRange(range);
+        scanner.forEach(entry -> {
+          String strKey = entry.getKey().getRow().toString();
+          assertFalse(expected.containsKey(strKey));
+          String strValue = entry.getValue().toString();
+          actualOldStyle.put(strKey, strValue);
+        });
+        assertEquals(expected.size(), actualOldStyle.size());
+        assertNotEquals(expected, actualOldStyle);
+      }
+
       upgrader.upgradeFileDeletes(getServerContext(), level);
 
       sleepUninterruptibly(1, TimeUnit.SECONDS);
-      Range range = MetadataSchema.DeletesSection.getRange();
-      Scanner scanner;
-      try {
-        scanner = c.createScanner(level.metaTable(), Authorizations.EMPTY);
-      } catch (TableNotFoundException e) {
-        throw new RuntimeException(e);
+      try (Scanner scanner = c.createScanner(level.metaTable(), Authorizations.EMPTY)) {
+        Map<String,String> actualNewStyle = new HashMap<>();
+        scanner.setRange(range);
+        scanner.forEach(entry -> {
+          String strKey = entry.getKey().getRow().toString();
+          String expectedValue = expected.get(strKey);
+          assertNotNull(expectedValue);
+          String strValue = entry.getValue().toString();
+          assertEquals(expectedValue, strValue);
```

Comment:
Will this be checked in the later check that the maps are equal?

Reply:
Yes, but... if there's an error, I'd rather see it fail here, where I don't have to view and determine the difference between two maps, each of size 100,000. The later check for the entire map equality is more of an extra sanity check on the whole list.